tests/src/authorization-tests.ts

```diff
@@ -82,43 +93,56 @@ test('Single sig spending condition uncompressed', () => {
+
+  const spendingCondition = createMultiSigSpendingCondition(
+    addressHashMode,
+    2,
+    pubKeys,
+    nonce,
+    fee
   );
-  spendingCondition.pubKeyEncoding = PubKeyEncoding.Compressed;
+  spendingCondition.signer = '11'.repeat(20);
 
-  const signature = new MessageSignature('fe'.repeat(65));
-  spendingCondition.signature = signature;
+  const signature = createMessageSignature('ff'.repeat(65));
+  const fields = [signature, signature, createStacksPublicKey(pubKeys[2])];
+  spendingCondition.fields = fields.map(createTransactionAuthField);
 
-  const serializedSpendingCondition = spendingCondition.serialize();
+  const serializedSpendingCondition = serializeSpendingCondition(spendingCondition);
 
   // prettier-ignore
   const spendingConditionBytesHex = [
     // address hash mode
-    AddressHashMode.SerializeP2WPKH,
+    AddressHashMode.SerializeP2SH,
     // signer
     0x11, 0x11, 0x11, 0x11, 0x11, 0x11, 0x11, 0x11, 0x11, 0x11, 0x11, 0x11, 0x11, 0x11, 0x11, 
     0x11, 0x11, 0x11, 0x11, 0x11,
     // nonce
-    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01, 0x59,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x7b,
     // fee rate
     0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01, 0xc8,
-    // key encoding
-    PubKeyEncoding.Compressed,
-    // signature
-    0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 
-    0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 
-    0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 
-    0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe,
-    0xfe, 0xfe, 0xfe, 0xfe, 0xfe
+    // length prefixed list of fields
+    0x00, 0x00, 0x00, 0x03,
+    0x02, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 
+    0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 
+    0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 
+    0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+    0xff, 0xff, 0xff, 0xff, 0xff,
+    0x02, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 
+    0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 
+    0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 
+    0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+    0xff, 0xff, 0xff, 0xff, 0xff,
+    0x00, 0x02, 0x02, 0x02, 0x02, 0x02, 0x02, 0x02, 0x02, 0x02, 0x02, 0x02, 0x02, 0x02, 0x02, 0x02, 0x02, 0x02, 0x02, 0x02, 0x02, 0x02, 0x02, 0x02, 0x02, 0x02, 0x02, 0x02, 0x02, 0x02, 0x02, 0x02, 0x02, 0x02,
+    // required signatures
+    0x00, 0x02
   ];
+
   const spendingConditionBytes = Buffer.from(spendingConditionBytesHex);
```

Comment:
Are there any test vectors copied from the rust codebase to ensure consistency?

Reply:
The multi-sig test in `transaction-tests.js` is mostly copied from the rust codebase, but I'm planning on adding some more test vectors. Will try and add some to this PR tomorrow